server/routerlicious/packages/services/src/dbFactory.ts

```diff
@@ -0,0 +1,55 @@
+/*!
+ * Copyright (c) Microsoft Corporation and contributors. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+import { IDbFactory } from "@fluidframework/server-services-core";
+import debug from "debug";
+import { Provider } from "nconf";
+import * as services from "./";
+
+interface IDBFactoryConfig {
+    name: string;
+    modulePathOrName: string;
+    dbFactoryConstructorName: string;
+    dbConfig: any;
+}
+
+const creator = (dbFactoryConfig: IDBFactoryConfig) => async (): Promise<IDbFactory> => {
+    debug(`Using ${dbFactoryConfig.name} Database`);
+    // Dynamically load the module
+    const externalDbFactoryModule = await import (dbFactoryConfig.modulePathOrName);
+    return new externalDbFactoryModule[dbFactoryConfig.dbFactoryConstructorName](
+        dbFactoryConfig.dbConfig,
+    ) as Promise<IDbFactory>;
+};
+
+export async function getDbFactory(config: Provider): Promise<IDbFactory> {
+    const dbFactoryConfig = config.get("db") as IDBFactoryConfig;
+    // Default handling is Mongo
+    if (!dbFactoryConfig || dbFactoryConfig.name === "mongo") {
+        const mongoFactory = new services.MongoDbFactory(config.get("mongo"));
+        return mongoFactory;
+    }
+
+    // We enable extensions loading only when the flag is set to true.
+    // The idea behind the flag to have an additional security measure to avoid loading unwanted extensions.
+    // @TODO: in the initial proposal it was part of the config file, but perhaps it should be an
```

Comment:
I am curious about this Todo item. Why might we want this to be an env variable instead of a config as it is now?